Lesson 2 homework

RamzanBersanov opened the pull request.

RamzanBersanov committed:
На ветке lesson2
 Изменения, которые будут включены в коммит:
	изменено:      README.md

RamzanBersanov committed:
На ветке lesson2
 Ваша ветка обновлена в соответствии с «origin/lesson2».
 Изменения, которые будут включены в коммит:
	изменено:      config/application.rb
	новый файл:    config/version.rb
	новый файл:    public/about.html
	новый файл:    public/about/author.html

VidgarVii commented on README.md:
```diff
@@ -1,24 +1,13 @@
+Аутентификация
+Базовое разграничение прав доступа
+Выполнение тестов
+Публикация результатов в соцсети
```
список сущностей приложения обычно выводится из сценария использования приложения

сценарий использования приложения примерно такой:
- Пользователь заходит на сайт
- он выбирает Тест
- Тест может принадлежать к одной из Категорий
- Тест состоит из Вопросов
- на каждый Вопрос есть несколько Ответов
- Пользователь выбирает один или несколько Ответов, чтобы ответить на каждый из Вопросов
- Результат тестирования сохраняется

Получается сущности это:
Категория, Тест, Вопрос и т.д

VidgarVii commented on config/application.rb:
```diff
@@ -6,10 +6,14 @@
-    config.load_defaults 7.0
+    config.i18n.default_locale = :ru
+
+    config.i18n.load_path += Dir[Rails.root.join('config', 'locales', '**', '*.{rb,yml}')]
```
что делает эта настройка? Она точно нужна?

RamzanBersanov replied:
какая? которая в 14 строке?

VidgarVii replied:
да, комментарий всегда пишется над кодом, на которой надо сделать исправление

VidgarVii commented on config/application.rb:
```diff
@@ -6,10 +6,14 @@
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 7.0
+    config.i18n.default_locale = :ru
```
чтобы не было ошибок в системе, надо еще добавить файл конфигураций в папку config/locales ru.yml. Содержание почти аналогичное как en.yml

VidgarVii commented on config/version.rb:
```diff
@@ -0,0 +1,5 @@
+module TestGuru
+  class Application
+    VERSION = "0.1.0"
```
"мутабельные" объекты (строки, хэши, массивы) рекомендуется `freeze`-ить, чтобы они были неизменяемыми
в противном случае константу, ссылающуюся на "мутабельный" объект,  можно изменить: `VERSION << '3.14..'`

VidgarVii commented on public/about.html:
```diff
@@ -0,0 +1,10 @@
+    <title>About project</title>
+  </head>
+  <body>
+    <h2>Проект второго модуля обучающего курса Ruby on Rails от школы Thinknetica. Подробнее о школе - на сайте <a href="http://www.thinknetica.com/">www.thinknetica.com</a></h2>
```
ссылки на сторонние ресурсы правильно открывать в новой вкладке и хорошо бы добавить опцию rel=nofollow rel=noopener. Так клиента сохраним и поискового робота дальше не пустим

VidgarVii commented on public/about.html:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
```
не стыковка с языком. Ни же то текст на русском. А код говорит, что английский

плохая идея делать отступы за счёт параграфов. Грубое нарушение семантики

RamzanBersanov replied:
не понял, где я сделал отступы?

VidgarVii replied:
это я случайно лишние скопипастил

RamzanBersanov committed:
На ветке lesson2
 Изменения, которые будут включены в коммит:
	изменено:      README.md
	изменено:      config/application.rb
        новый файл:    config/locales/ru.yml
	изменено:      config/version.rb
	изменено:      public/about.html
	изменено:      public/about/author.html
